.github/workflows/release.yml

```diff
@@ -21,38 +21,48 @@ jobs:
 
     steps:
       # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           repository: wingtk/gvsbuild
           fetch-depth: 0
 
       - name: Set up Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python }}
           architecture: ${{ matrix.arch }}
+          cache: 'pip'
+          cache: 'poetry'
 
       - name: Install Build Deps
         run: python -m pip install wheel
 
+      - name: Install Poetry
+        run: python -m pip install poetry==1.1.13
+      - name: Configure Poetry
+        run: poetry config virtualenvs.in-project true
+      - name: Install Python Dependencies
+        run: poetry install
+
       - name: Move git binary
         # Temporarily move preinstalled git to prevent errors related to cygwin.
         run: Rename-Item "C:\Program Files\Git\usr\bin" notbin
 
+      - name: Build gvsbuild
+        run: pip install .
+
       - name: Build
         run: >
-          python build.py build
+          gvsbuild build
           --platform=${{ matrix.arch }}
           --vs-ver=${{ matrix.vstudio }}
           --same-python
           --enable-gi
           --py-wheel
-          --skip gtksourceview,emeus,clutter
+          --patches-root-dir=${{ github.workspace }}\patches
+          --skip gtksourceview4,emeus,clutter
           gtk3-full pycairo pygobject lz4 enchant
 
-      - name: Restore git binary
-        run: Rename-Item  "C:\Program Files\Git\usr\notbin" bin
-
```

Comment:
I think we should retain this step. What was the reason for removing?

Reply:
Yes it does depend but in this situation we want to ensure that since we made the change in the first place we also restore the system back to it's original state so that other steps that might use it don't break.